### PR TITLE
Added cart features

### LINKED ISSUE #1657: Implement complete cart functionalities.
Currently in the website the cart is not working.

I can do the following changes.

1. Add items to cart when user clicks on add to cart button. If item already exists quantity will be increased.
2. In the cart page selected items will be listed. 
3. Users can update (+/-) the quantity.
4. Users can remove an item from the cart.

There are 2 options to implement cart feature. One is using local storage. Other one is storing in server.

I will be okay with both options.  Please let me know, which method you would like to proceed with this approach or if you have any further suggestions.

### PR DESCRIPTION
**Pull request by DharshiBalasubramaniyam**
Fixes Issue
This pr fixes #1657 

I have closed the existing pr and creating this new pr. I have created a new pr because the previous one showed some deployment errors. Now, I have used react redux for the state management. 

Changes proposed
The following changes have been implemented:

Add to Cart Functionality:
When users click the "Add to Cart" button, items are added to the cart.
If the item already exists in the cart, its quantity is increased.
Cart Page Enhancements:

Selected items are now listed on the cart page.
Users can update the quantity of items (+/-).
Users have the option to remove items from the cart.
User can clear the cart as well.

Guest Cart Experience:
Users can add items to the cart without logging in.
Cart items are stored in the user's browser to enhance the user experience.
Users must log in before proceeding to checkout.

## Summary by CodeRabbit

- **New Features**
  - Introduced Redux for enhanced state management across the app.
  - Display the number of items in the cart dynamically on the cart icon.
  - Improved cart functionality with dynamic item addition and removal using Redux.
  - Introduced conditional components for checkout based on user login status.

- **Improvements**
  - Removed conditional rendering of the cart icon in the Navbar for consistent user experience.
  - Enhanced the layout and text display in the empty cart view for better user interface.

- **Dependencies**
  - Added `@reduxjs/toolkit`, `react-hot-toast`, and `react-redux` to the project dependencies.

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -24,7 +24,6 @@ import Signup from "./pages/Signup/Signup";
 import FAQ from "./pages/FAQ/Faq";
 import CartEmpty from "./pages/Order/CartEmpty";
 import PrivateRoute from "./PrivateRoute";
-import TermsAndConditions from "./pages/TermsAndCondition/TermsAndCondition";
```
Comment (codervivek5) [inline src/App.jsx]: Why remove this line ?

File: src/pages/Order/Cart.jsx
Comment (codervivek5) [inline src/pages/Order/Cart.jsx:1]: Instead of hardcore product .... Kindly try with fakeapi product for demo purpose

**Comment by codervivek5** [comment]
Also share any image or video of new changes using Redux toolkit

File: src/App.jsx (hunk 1 above)
Comment (DharshiBalasubramaniyam) [reply src/App.jsx]: solved!

File: src/pages/Order/Cart.jsx
Comment (DharshiBalasubramaniyam) [reply src/pages/Order/Cart.jsx:1]: I have not used hardcore products anywhere. I did not get you.

**Comment by DharshiBalasubramaniyam** [comment]
> Also share any image or video of new changes using Redux toolkit


https://github.com/codervivek5/VigyBag/assets/139672976/b5b807bb-f021-42d8-b825-1e1d23a82c40


https://github.com/codervivek5/VigyBag/assets/139672976/c6079afa-702f-48dd-96b6-c6f65b5fa17d

**Comment by codervivek5** [comment]
Need to make some changes. Are you connected in discord if not kindly join them ping me ....will discuss via meet

File: src/pages/Order/Cart.jsx [hunk 2]
```diff
@@ -104,106 +105,103 @@ const Subtotal = ({ items }) => {
...
+  const cartItems = useSelector(state => state.cart.items)
+  const dispatch = useDispatch()
+
+  const onUpdate = (product, quantity) => {
+    dispatch(manageCartItem({ product, quantity }))
+  }
+
+  const onclear = () => {
+    dispatch(clearCart())
+  }
```
[the remark this answers is not in the document]
Comment (codervivek5) [reply src/pages/Order/Cart.jsx]: Chec this review

File: src/redux/cartSlice.jsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+                    price: product.price,
+                    total: newQuantity * product.price,
+                    image: product.image,
+                }
+                newCart = [...filteredCart, newItem]
+            }
+            newCart = newCart.sort((a, b) => a.id - b.id)
+            localStorage.setItem("vigybag-cart", JSON.stringify(newCart))
+            state.items = newCart
+        },
```
[the remark this answers is not in the document]
Comment (codervivek5) [reply src/redux/cartSlice.jsx]: Chck this review also

**Comment by DharshiBalasubramaniyam** [comment]
After changes: 

https://github.com/codervivek5/VigyBag/assets/139672976/d06ff159-8343-4156-9f7a-b4f62ea7efe3

image

**Pull request merged**